src/main/java/com/airbnb/kafka/KafkaStatsdMetricsReporter.java

```diff
@@ -23,119 +21,130 @@
 
 /**
- * Kafka uses Yammer Metrics for metrics reporting in both the server and the client.
- * This can be configured to report stats using pluggable stats reporters to hook up to your monitoring system.
- *
- * This class allows to report Kafka Metrics to StatsD.
- *
- * The reporter is enabled and configured from the Kafka config file, server.properties.
  *
- * @see kafka.metrics.KafkaMetricsReporter
  */
 public class KafkaStatsdMetricsReporter implements KafkaStatsdMetricsReporterMBean, KafkaMetricsReporter {
 
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(StatsDReporter.class);
 
-    public static final String DEFAULT_EXCLUDE_REGEX = "(kafka\\.consumer\\.FetchRequestAndResponseMetrics.*)|(.*ReplicaFetcherThread.*)|(kafka\\.server\\.FetcherLagMetrics\\..*)|(kafka\\.log\\.Log\\..*)|(kafka\\.cluster\\.Partition\\..*)";
+    public static final String DEFAULT_REGEX_FILTER = "(kafka\\.consumer\\.FetchRequestAndResponseMetrics.*)|(.*ReplicaFetcherThread.*)|(kafka\\.server\\.FetcherLagMetrics\\..*)|(kafka\\.log\\.Log\\..*)|(kafka\\.cluster\\.Partition\\..*)";
 
-    private AbstractPollingReporter underlying;
-    private boolean enabled = false;
-    private boolean running = false;
-    private final Logger logger = Logger.getLogger(getClass());
+    @Override
+    public String getMBeanName() {
+        return "kafka:type=" + getClass().getName();
+    }
+
+    private boolean enabled;
+    private AtomicBoolean running = new AtomicBoolean(false);
     private String host;
     private int port;
     private String prefix;
+    private long pollingPeriodInSeconds;
     private MetricDimensionOptions metricDimensionOptions;
     private MetricPredicate metricPredicate;
-
-
-    @Override
-    public String getMBeanName() {
-        return "kafka:type="+getClass().getName();
-    }
+    private StatsDClient statsd;
+    private boolean isTagSupported;
+    private AbstractPollingReporter underlying = null;
 
     public boolean isRunning() {
-        return running;
+        return running.get();
     }
 
+    //try to make it compatible with kafka-statsd-metrics2
     @Override
     public synchronized void init(VerifiableProperties props) {
+        loadConfig(props);
+        if (enabled) {
+            log.info("Reporter is enabled and starting...");
+            startReporter(pollingPeriodInSeconds);
+        } else {
+            log.warn("Reporter is disabled");
+        }
+    }
+
+    private void loadConfig(VerifiableProperties props) {
         enabled = props.getBoolean("external.kafka.statsd.reporter.enabled", false);
-        int pollingIntervalSecs = props.getInt("kafka.metrics.polling.interval.secs", 10);
         host = props.getString("external.kafka.statsd.host", "localhost");
         port = props.getInt("external.kafka.statsd.port", 8125);
-        prefix = props.getString("external.kafka.statsd.metrics.prefix", null);
-        metricPredicate = MetricPredicate.ALL;
```

Comment:
`metricPredicate` does not seem to have a default value anymore.
will that cause NPE?